api/src/main/java/me/lokka30/treasury/api/economy/account/AccountData.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.economy.account;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents a data object, holding an account data.
+ *
+ * @author MrIvanPlays
+ * @since 2.0.0
+ */
+public final class AccountData {
+
+    /**
+     * Creates a new {@link PlayerAccount player account} data.
+     *
+     * @param uniqueId the unique id of the player account owner
+     * @return new account data instance
+     * @since 2.0.0
+     */
+    @NotNull
+    public static AccountData forPlayerAccount(@NotNull UUID uniqueId) {
+        return new AccountData(
+                true,
+                Optional.empty(),
+                Optional.of(Objects.requireNonNull(uniqueId, "uniqueId"))
+        );
+    }
+
+    /**
+     * Creates a new {@link NonPlayerAccount non player account} data.
+     *
+     * @param identifier the identifier of the non player account
+     * @return new account data instance
+     * @since 2.0.0
+     */
+    @NotNull
+    public static AccountData forNonPlayerAccount(@NotNull String identifier) {
+        return new AccountData(
+                false,
+                Optional.of(Objects.requireNonNull(identifier, "identifier")),
+                Optional.empty()
+        );
+    }
+
+    private final boolean player;
+    private final Optional<String> identifier;
+    private final Optional<UUID> uniqueId;
+
+    private AccountData(boolean player, Optional<String> identifier, Optional<UUID> uniqueId) {
```

Comment:
Specify nullability of these parameters. (The optionals should be `@NotNull`)

**OR**

It seems redundant to use Optional here, since it has `private` access. Optional is used when we are worried that it is likely a certain nullable parameter will be mistaken for non-null. We are the ones using the method so using nullable parameters instead will suffice.

Reply:
> Specify nullability of these parameters.

why? it's a private constructor. doesn't really matter

> It seems redundant to use Optional here, since it has `private` access. Optional is used when we are worried that it is likely a certain nullable parameter will be mistaken for non-null. We are the ones using the method so using nullable parameters instead will suffice.

this is in order to not allocate an optional each time the getters are used.

Reply:
> why? it's a private constructor. doesn't really matter

Public or private, it is important to specify the nullability of method parameters since it forms a part of documentation and no assumptions need to be made. They're not strictly necessary, but they're so easy to implement, so they may as well go in.

> this is in order to not allocate an optional each time the getters are used.

👍🏻